File: ast/src/testing/vanila/go_scripts/models/post.go
package models

import (
	"errors"
	"time"

	"go_scripts/utils"
)

type Post struct {
	ID        string    `json:"id"`
	AuthorID  string    `json:"author_id"`
	Title     string    `json:"title"`
	Content   string    `json:"content"`
	Published bool      `json:"published"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

var ErrEmptyContent = errors.New("post content cannot be empty")

func NewPost(authorID, title, content string) (*Post, error) {
	if content == "" {
		return nil, ErrEmptyContent
	}
	
	id, err := utils.GenerateID()
	if err != nil {
		return nil, err
	}

	now := time.Now()
	return &Post{
		ID:        id,
		AuthorID:  authorID,
		Title:     title,
		Content:   content,
		Published: false,
		CreatedAt: now,
		UpdatedAt: now,
	}, nil
}

File: ast/src/testing/vanila/typescript_bare/models/Post.ts
import { IEntity } from "../types.ts";

export class Post implements IEntity {
  id: string;
  authorId: string;
  title: string;
  content: string;
  createdAt: Date;
  updatedAt: Date;

  constructor(data: Partial<Post>) {
    this.id = data.id || "";
    this.authorId = data.authorId || "";
    this.title = data.title || "";
    this.content = data.content || "";
    this.createdAt = data.createdAt || new Date();
    this.updatedAt = data.updatedAt || new Date();
  }
}
